azurerm/internal/services/compute/virtual_machine.go

```diff
@@ -411,3 +416,407 @@ func flattenVirtualMachineOSDisk(ctx context.Context, disksClient *compute.Disks
 		},
 	}, nil
 }
+
+func virtualMachineDataDiskSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"local": virtualMachineLocalDataDiskSchema(),
```

Comment:
part of me thinks this should be called `new` rather than `local`?

Reply:
As discussed offline, these have been updated to `create` and `attach` to better describe their relationship to the VM.